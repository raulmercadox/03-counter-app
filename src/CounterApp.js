import { Fragment, useState } from "react";

const CounterApp = ({ valor = 100 }) => {
  const [counter, setCounter] = useState(valor);

  const incrementHandler = () => {
    setCounter((prev) => prev + 1);
  };

  const decrementHandler = () => {
    setCounter((prev) => prev - 1);
  };

  const resetHandler = () => {
    setCounter(valor);
  };

  return (
    <Fragment>
      <label>{counter}</label>
      <button onClick={incrementHandler}>+</button>
      <button onClick={decrementHandler}>-</button>
      <button onClick={resetHandler}>Reset</button>
    </Fragment>
  );
};

export default CounterApp;
