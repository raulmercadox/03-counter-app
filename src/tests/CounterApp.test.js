import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en CounterApp", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("Debe mostrar <CounterApp/> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el valor por defecto de 100", () => {
    const valorDefecto = wrapper.find("label").text();
    expect(valorDefecto).toBe("100");
  });

  test("Debe visualizar un valor de 350", () => {
    const valorInicial = 350;
    const wrapper = shallow(<CounterApp valor={valorInicial} />);
    const valorObtenido = wrapper.find("label").text();
    expect(valorObtenido).toBe(String(valorInicial));
  });

  test("Prueba de boton que aumenta en 1", () => {
    wrapper.find("button").at(0).simulate("click");
    const valorObtenido = wrapper.find("label").text();
    expect(valorObtenido).toBe("101");
  });

  test("Prueba de boton que disminuye en 1", () => {
    wrapper.find("button").at(1).simulate("click");
    const valorObtenido = wrapper.find("label").text();
    expect(valorObtenido).toBe("99");
  });

  test("Probar el boton reset", () => {
    const valorInicial = 350;
    const wrapper = shallow(<CounterApp valor={valorInicial} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(2).simulate("click"); // Bot√≥n reset
    const valorObtenido = wrapper.find("label").text();
    expect(valorObtenido).toBe(String(valorInicial));
  });
});
