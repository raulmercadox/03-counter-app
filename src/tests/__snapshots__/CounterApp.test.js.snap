// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe mostrar <CounterApp/> correctamente 1`] = `
<Fragment>
  <label>
    100
  </label>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
